lib/VMwareWebService/broker_timeout.rb

```diff
@@ -0,0 +1,64 @@
+module Timeout
```

Comment:
Cab you also please add a big comment at the top of this explaining the need for overriding Timeout, the problem you are trying to solve, how this solves it, and maybe a TODO note to revisit this after we get to Ruby 2.2 to see if there is an underlying Ruby thread.c bug?  @tenderlove @matthewd Was there anything else you wanted to see in the comment?